### PR TITLE
Enhance Build Process

### PR DESCRIPTION
**Pull request by laardee**
This PR introduces some improvements and optimizations:

* Splits Gulp runtime configurations into router-specific files for better modularity.
* Updates Dockerfiles, light-http configs, and scripts to use parameters, avoiding unnecessary changes to git-tracked files.
* Improves the clarity of error messages in specific scenarios.
* Adds support for environment variables to skip selected build steps, reducing development time.
* Skips pushing images to the Docker registry if credentials are not defined, ensuring the build process in unconfigured environments, like development.

Partly related to the old issue #189

### CODE REVIEW

File: task/Common.js
Comment (vesameskanen) [inline task/Common.js]: Would it be more natural to import this one liner from util.js ? task folder contains gulp tasks and the main update logic.

**Review by vesameskanen — CHANGES_REQUESTED** [review]
Thank you for the contribution. Using variables instead of modifying local  version managed files is a good improvement. 

Some notes, though:

- Unused task/PatchDeploymentFiles.js should be deleted, the reference to it in gulpfile removed and  README.md updated to reflect the changes.

- Build process works but creates a broken OTP image.  When running the image,  kubernetes deployment shows an error:

    Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       StartError
      Message:      failed to create containerd task: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "/var/otp/entrypoint.sh": stat /var/otp/entrypoint.sh: no such file or directory: unknown

This error is a bit strange as I can find /var/otp/entrypoint.sh from the docker image. Any ideas?

**Comment by vesameskanen** [comment]
Found out that our container deployment mounts data storage to path  `/var/otp` .  I assume that this mounting overwrites the original folder where the entrypoint script is.

File: task/Common.js
Comment (laardee) [reply task/Common.js]: True, moved that to util.js

**Comment by laardee** [comment]
> Found out that our container deployment mounts data storage to path `/var/otp` . I assume that this mounting overwrites the original folder where the entrypoint script is.

The entrypoint.sh is now in the parent `/var/` directory.

Are these mounts and other cluster-related variables documented somewhere? I briefly tried to find those without success.

**Comment by vesameskanen** [comment]
Opentripplanner-data-container, opentripplanner and their deployment configurations are indeed a complex system. It is hard to figure out how the actual runtime environment and configurations found from it get assembled, because the logic is split into many repositories.

The mentioned conflicting mount path is defined here:
https://github.com/HSLdevcom/digitransit-kubernetes-deploy/blob/master/roles/aks-apply/files/dev/opentripplanner-hsl-v2-dev.yml#L58

Environment variables for the data loader can be found here:
https://github.com/HSLdevcom/digitransit-kubernetes-deploy/blob/master/roles/aks-apply/files/dev/otp-data-builder-hsl-v3-dev.yml

These deployments are for HSL routing in development environment.

**Review by vesameskanen — APPROVED** [review]
Seems to work now.  I will ask second opinion from my colleague Joel before merging.

**Pull request merged**
